### ISSUE TITLE
Segfault on reindex

### ISSUE THREAD

**Issue by underdarkskies** [RavenProject/Ravencoin#208]
After performing an initial sync on Mainnet,
I enable txindex in raven.conf
I restart the client and it warns that a reindex is necessary
I restart the daemon with `-reindex` flag set
segfault occurs
![image](https://user-images.githubusercontent.com/35243703/43931641-7dc1d160-9c0d-11e8-9b1e-759b815ba05d.png)

Debug.log showing the crash.
https://pastebin.com/UqD5K8jd

**Comment by underdarkskies** [comment]
raven.conf: 
```
reindex=1
server=1
whitelist=127.0.0.1
rpcport=18766
rpcallowip=127.0.0.1
rpcuser=ravencoin
rpcpassword=test1234
```

**Comment by spyder46n2** [comment]
I'm trying to reproduce this right now.  I have an AWS Ubuntu instance with the latest Master branch (2.0.3).  I didn't have any block info or wallet data.  Right now I'm syncing, once that is finished I'll match your conf file and see if I get the same issue.

**Comment by underdarkskies** [comment]
gdb output 
```
(gdb) frame 0
#0  CBlockIndex::GetMedianTimePast (this=<optimized out>) at chain.h:319
319                 *(--pbegin) = pindex->GetBlockTime();
(gdb) frame 1
#1  CBlockIndex::GetMedianTimePast() const () at miner.cpp:63
63          int64_t nNewTime = std::max(pindexPrev->GetMedianTimePast()+1, GetAdjustedTime());
(gdb) frame 2
#2  0x00005555557a8f70 in ContextualCheckBlock (block=..., state=..., consensusParams=..., pindexPrev=<optimized out>, assetCache=0x555556f0d3f0) at validation.cpp:3510
3510                                  ? pindexPrev->GetMedianTimePast()
(gdb) frame 3
#3  0x00005555557b5225 in AcceptBlock (pblock=..., state=..., chainparams=..., ppindex=ppindex@entry=0x0, fRequested=fRequested@entry=true, dbp=dbp@entry=0x7fffdf27cbb0, fNewBlock=0x0) at validation.cpp:3702
3702            !ContextualCheckBlock(block, state, chainparams.GetConsensus(), pindex->pprev, passets)) {
(gdb) frame 4
#4  0x00005555557cce80 in LoadExternalBlockFile (chainparams=..., fileIn=fileIn@entry=0x7fffd8000c10, dbp=dbp@entry=0x7fffdf27cbb0) at validation.cpp:4606
4606                        if (AcceptBlock(pblock, state, chainparams, nullptr, true, dbp, nullptr)) {
(gdb) frame 5
#5  0x00005555555e353d in ThreadImport (vImportFiles=...) at init.cpp:652
652                 LoadExternalBlockFile(chainparams, file, &pos);
```

**Comment by russkidooski** [comment]
I got this issue a week ago but didn't think anything about it, I'm going to try and reproduce the issue and dump my debug here.

**Comment by underdarkskies** [comment]
@spyder46n2 a full sync isnt necessary to see the segfault in action. gracefully shut down the daemon and reindex

**Comment by russkidooski** [comment]
Here's my debug log: https://pastebin.com/pnfE91PD
Windows 10 using Raven-qt version 2.0.2

**Comment by underdarkskies** [comment]
```
(gdb) backtrace
#0  CBlockIndex::GetMedianTimePast (this=<optimized out>) at chain.h:319
#1  CBlockIndex::GetMedianTimePast() const () at miner.cpp:63
#2  0x00005555557a8f70 in ContextualCheckBlock (block=..., state=..., consensusParams=..., pindexPrev=<optimized out>, assetCache=0x555556f0d3f0) at validation.cpp:3510
#3  0x00005555557b5225 in AcceptBlock (pblock=..., state=..., chainparams=..., ppindex=ppindex@entry=0x0, fRequested=fRequested@entry=true, dbp=dbp@entry=0x7fffdf27cbb0, fNewBlock=0x0) at validation.cpp:3702
#4  0x00005555557cce80 in LoadExternalBlockFile (chainparams=..., fileIn=fileIn@entry=0x7fffd8000c10, dbp=dbp@entry=0x7fffdf27cbb0) at validation.cpp:4606
#5  0x00005555555e353d in ThreadImport (vImportFiles=...) at init.cpp:652
#6  0x0000555555603b54 in boost::_bi::list1<boost::_bi::value<std::vector<boost::filesystem::path, std::allocator<boost::filesystem::path> > > >::operator()<void (*)(std::vector<boost::filesystem::path, std::allocator<boost::filesystem::path> >), boost::_bi::list0> (f=<optimized out>, a=<synthetic pointer>..., this=<optimized out>) at /home/ubuntu/ravencoin/depends/x86_64-linux-gnu/share/../include/boost/bind/bind.hpp:259
#7  boost::_bi::bind_t<void, void (*)(std::vector<boost::filesystem::path, std::allocator<boost::filesystem::path> >), boost::_bi::list1<boost::_bi::value<std::vector<boost::filesystem::path, std::allocator<boost::filesystem::path> > > > >::operator() (this=<optimized out>) at /home/ubuntu/ravencoin/depends/x86_64-linux-gnu/share/../include/boost/bind/bind.hpp:1294
#8  boost::detail::thread_data<boost::_bi::bind_t<void, void (*)(std::vector<boost::filesystem::path, std::allocator<boost::filesystem::path> >), boost::_bi::list1<boost::_bi::value<std::vector<boost::filesystem::path, std::allocator<boost::filesystem::path> > > > > >::run (this=<optimized out>) at /home/ubuntu/ravencoin/depends/x86_64-linux-gnu/share/../include/boost/thread/detail/thread.hpp:116
#9  0x0000555555a6cd8d in thread_proxy ()
#10 0x00007ffff7bbd6db in start_thread (arg=0x7fffdf27d700) at pthread_create.c:463
#11 0x00007ffff712888f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by underdarkskies** [comment]
To be perfectly clear:
This occurred with a binary compiled from the depends system, which uses boost 1.64

 I have not tested the native bionic repo of 1.65.1 yet

Xenial ships with 1.58 .

**Comment by bedri** [comment]
Issue continues on 2.0.3 testnet4

New sub asset creation from raven-cli were shown conflicted in testnet4 but they show up in listassets so I wanted to reindex

```
raven-qt -testnet -server -daemon -reindex
Segmentation fault (core dumped)
```
And continues to give Segmentation fault on running without -reindex

Now deleted blocks and syncing from scratch and will check both conflicted transactions and reindex.

OS: OpenSuSE Tumbleweed
libboost version 1.67

**Comment by TronBlack** [comment]
Fixed in 2.0.4
